File: LeetCodeTasks/3000_maximum-area-of-longest-diagonal-rectangle/solutions/cpp/solution.cpp
#include <vector>
#include <cmath>
using namespace std;

class Solution {
public:
    int areaOfMaxDiagonal(vector<vector<int>>& dimensions) {
        int maxArea = 0;
        double longestDiagonal = 0;

        for (const auto& dim : dimensions) {
            int length = dim[0];
            int width = dim[1];
            double diagonal = sqrt(length * length + width * width);
            int area = length * width;
            
            if (diagonal > longestDiagonal) {
                longestDiagonal = diagonal;
                maxArea = area;
            } else if (diagonal == longestDiagonal) {
                if (area > maxArea) {
                    maxArea = area;
                }
            }
        }
        
        return maxArea;
    }
};

File: LeetCodeTasks/3000_maximum-area-of-longest-diagonal-rectangle/solutions/python3/solution.py
class Solution:
    def areaOfMaxDiagonal(self, dimensions: list[list[int]]) -> int:
        max_diagonal = 0
        max_area = 0
        
        for length, width in dimensions:
            diagonal = length ** 2 + width ** 2
            area = length * width
            
            if diagonal > max_diagonal or (diagonal == max_diagonal and area > max_area):
                max_diagonal = diagonal
                max_area = area
                
        return max_area
